pkg/database/database.go

```diff
@@ -55,6 +57,59 @@ func Reset(databasePath string) error {
 	return nil
 }
 
+func RunBackup(databasePath string) {
+	dbExists, _ := utils.FileExists(databasePath)
+	//if database exists (setup is completed)
+	//create a backup of the db if a migration is needed
+	if dbExists {
+		err := Backup(databasePath, false)
+
+		if err != nil {
+			logger.Fatalf("Backup error: %s", err)
+		}
+	}
+
+}
+
+// Backup the database if a migration is needed
+// or force is set to true
+func Backup(databasePath string, force bool) error {
+
+	db, err := sqlx.Connect(sqlite3Driver, "file:"+databasePath+"?_fk=true")
+	if err != nil {
+		return fmt.Errorf("Open database %s failed:%s", databasePath, err)
+	}
+	defer db.Close()
+
+	var dbVersion uint
+	err = db.Get(&dbVersion, "SELECT version FROM schema_migrations")
+	if err != nil {
+		return err
+	}
+
+	if dbVersion > appSchemaVersion {
```

Comment:
`runMigrations` already gets the database versions and checks the difference. Can we not call this stuff from there? We can ensure that backup doesn't run on new databases by checking for schema version `0`.

Reply:
i made it independent so that it might be called from elsewhere for future purely backup purposes
eg task -> backup database

Reply:
Also in runMigrations as it is now you don't have access to the database to run a query.
migrate.New is used to directly access the db and it only returns a migrate type.
If we wanted to do it there we would need to add the sql.Open there , check for errors , run the migrate.NewWithDatabaseInstance and so on ....